### ISSUE TITLE
"free(): invalid pointer" on calling torch.device('any_device')

### ISSUE THREAD

**Issue by Piteryo** [jupyter-xeus/xeus-python#405]
Running cell with ```torch.device()``` call causes crash of the kernel with the following error: "free(): invalid pointer"

Happens on the latest version from conda with jupyter lab 3.0.7 and PyTorch 1.6

**Comment by SylvainCorlay** [comment]
Hey @Piteryo , thanks for posting.

Is this issue related to xeus-python? Does `torch.device()` work in the same environment besides xeus-python?

**Comment by Piteryo** [comment]
> Hey @Piteryo , thanks for posting.
> 
> Is this issue related to xeus-python? Does `torch.device()` work in the same environment besides xeus-python?

Yes, there is no such problem when using the default python kernel in the same environment.

**Comment by SylvainCorlay** [comment]
Would you mind posting the content of `conda list` in this environment?

**Comment by Piteryo** [comment]
<details>
  <summary>Conda list results</summary>
  # packages in environment at /opt/conda/envs/test_wtf:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main  
anyio                     2.0.2            py36h5fab9bb_4    conda-forge
async_generator           1.10                       py_0    conda-forge
attrs                     20.3.0             pyhd3deb0d_0    conda-forge
babel                     2.9.0              pyhd3deb0d_0    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                        py_2    conda-forge
backports.functools_lru_cache 1.6.1                      py_0    conda-forge
bleach                    3.3.0              pyh44b312d_0    conda-forge
ca-certificates           2020.12.5            ha878542_0    conda-forge
certifi                   2020.12.5        py36h5fab9bb_1    conda-forge
contextvars               2.4                        py_0    conda-forge
dataclasses               0.7                pyhe4b4509_6    conda-forge
decorator                 4.4.2                      py_0    conda-forge
defusedxml                0.6.0                      py_0    conda-forge
entrypoints               0.3             pyhd8ed1ab_1003    conda-forge
future                    0.18.2                   pypi_0    pypi
idna                      3.1                pyhd3deb0d_0    conda-forge
immutables                0.14             py36h8c4c3a4_1    conda-forge
importlib-metadata        3.4.0            py36h5fab9bb_0    conda-forge
importlib_metadata        3.4.0                hd8ed1ab_0    conda-forge
ipykernel                 5.4.3            py36he448a4c_0    conda-forge
ipython                   7.16.1           py36he448a4c_2    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
jedi                      0.17.2           py36h5fab9bb_1    conda-forge
jinja2                    2.11.3             pyh44b312d_0    conda-forge
json5                     0.9.5              pyh9f0ad1d_0    conda-forge
jsonschema                3.2.0                      py_2    conda-forge
jupyter_client            6.1.11             pyhd8ed1ab_1    conda-forge
jupyter_core              4.7.1            py36h5fab9bb_0    conda-forge
jupyter_server            1.3.0            py36h5fab9bb_0    conda-forge
jupyterlab                3.0.7              pyhd8ed1ab_0    conda-forge
jupyterlab_pygments       0.1.2              pyh9f0ad1d_0    conda-forge
jupyterlab_server         2.2.0              pyhd8ed1ab_0    conda-forge
ld_impl_linux-64          2.33.1               h53a641e_7  
libedit                   3.1.20191231         h14c3975_1  
libffi                    3.3                  he6710b0_2  
libgcc-ng                 9.1.0                hdf63c60_0  
libsodium                 1.0.18               h36c2ea0_1    conda-forge
libstdcxx-ng              9.1.0                hdf63c60_0  
libuuid                   2.32.1            h14c3975_1000    conda-forge
markupsafe                1.1.1            py36he6145b8_2    conda-forge
mistune                   0.8.4           py36h1d69622_1002    conda-forge
nbclassic                 0.2.6              pyhd8ed1ab_0    conda-forge
nbclient                  0.5.1                      py_0    conda-forge
nbconvert                 6.0.7            py36h5fab9bb_3    conda-forge
nbformat                  5.1.2              pyhd8ed1ab_1    conda-forge
ncurses                   6.2                  he6710b0_1  
nest-asyncio              1.4.3              pyhd8ed1ab_0    conda-forge
notebook                  6.0.3            py36h9f0ad1d_1    conda-forge
numpy                     1.19.5                   pypi_0    pypi
openssl                   1.1.1h               h516909a_0    conda-forge
packaging                 20.9               pyh44b312d_0    conda-forge
pandoc                    2.11.4               h7f98852_0    conda-forge
pandocfilters             1.4.2                      py_1    conda-forge
parso                     0.7.1              pyh9f0ad1d_0    conda-forge
pexpect                   4.8.0              pyh9f0ad1d_2    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pip                       20.3.3           py36h06a4308_0  
prometheus_client         0.9.0              pyhd3deb0d_0    conda-forge
prompt-toolkit            3.0.14             pyha770c72_0    conda-forge
ptvsd                     4.3.2            py36h8c4c3a4_3    conda-forge
ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
pygments                  2.7.4              pyhd8ed1ab_0    conda-forge
pyparsing                 2.4.7              pyh9f0ad1d_0    conda-forge
pyrsistent                0.17.3           py36h1d69622_1    conda-forge
python                    3.6.12               hcff3b4d_2  
python-dateutil           2.8.1                      py_0    conda-forge
python_abi                3.6                     1_cp36m    conda-forge
pytz                      2021.1             pyhd8ed1ab_0    conda-forge
pyzmq                     20.0.0           py36h2b1bd32_1    conda-forge
readline                  8.1                  h27cfd23_0  
requests                  2.13.0                   py36_0    conda-forge
send2trash                1.5.0                      py_0    conda-forge
setuptools                52.0.0           py36h06a4308_0  
six                       1.15.0             pyh9f0ad1d_0    conda-forge
sniffio                   1.2.0            py36h5fab9bb_1    conda-forge
sqlite                    3.33.0               h62c20be_0  
terminado                 0.9.2            py36h5fab9bb_0    conda-forge
testpath                  0.4.4                      py_0    conda-forge
tk                        8.6.10               hbc83047_0  
torch                     1.6.0                    pypi_0    pypi
tornado                   6.1              py36h1d69622_0    conda-forge
traitlets                 4.3.3            py36h9f0ad1d_1    conda-forge
typing_extensions         3.7.4.3                    py_0    conda-forge
wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
webencodings              0.5.1                      py_1    conda-forge
wheel                     0.36.2             pyhd3eb1b0_0  
xeus                      0.24.4               h67b5559_0    conda-forge
xeus-python               0.8.6            py36h233d77c_1    conda-forge
xz                        5.2.5                h7b6447c_0  
zeromq                    4.3.3                h58526e2_3    conda-forge
zipp                      3.4.0                      py_0    conda-forge
zlib                      1.2.11               h7b6447c_3  

</details>

**Comment by Piteryo** [comment]
Conda list results:
https://pastebin.com/u5VqVk3a

**Comment by SylvainCorlay** [comment]
Thanks, you appear to have a mix of packages from the default channel and from PyPI which can be incompatible with conda-forge.

I recommend using only conda-forge packages.

A good means to achieve this is to start from a miniforge installer, which defaults to conda-forge: https://github.com/conda-forge/miniforge. (If you get the mambaforge installer, you even get mamba in the base environment), and to only install stuff from conda-forge.

PyPI wheels often vendor binaries which override installed packages, or may be binary incompatible.
